### DISCUSSION TITLE
Postgresql parameterized ~ query fails

### DISCUSSION THREAD

**Discussion by robocoder** [questdb/questdb#1378] [Ideas]
QuestDB: 6.0.7.1
Java: openjdk-11-jre-headless:amd64  11.0.11+9-0ubuntu2~20.04  
OS: Ubuntu 20.04 LTS
Postgresql client: 12.8-0ubuntu0.20.04.1
PHP: 8.0.11 with PDO, pdo_pgsql extensions

Example table:

```sql
CREATE TABLE foo (
    ts TIMESTAMP,
    code SYMBOL INDEX
) timestamp(ts) PARTITION BY DAY;
```

Query via psql (works):
```sql
SELECT code, count() FROM foo WHERE ts IN '2021-10-03' AND code ~ '154';
```

Parameterized query  via php (errors out):
```php
<?php
$tsn = '2021-10-03';
$pattern = '154';
$sql = <<<END_SQL
SELECT code, count() FROM foo WHERE ts IN ? AND code ~ ?
END_SQL
;
$dsn = 'pgsql:host=localhost;port=8812;dbname=qdb;user=admin;password=quest';
$dbh = new PDO($dsn);
$sth = $dbh->prepare($sql);
$sth->execute([$tsn, $pattern]);
```

Error:
```
2021-10-04T00:50:29.942524Z E i.q.c.p.PGConnectionContext error [pos=109, msg=`unexpected argument for function: ~. expected args: (STRING,STRING constant). actual args: (SYMBOL,unknown)`, errno=`0]
```

Do you think it's a questdb bug in the postresql wire protocol, or a php pdo_pgsql bug?

**Comment by pswu11** [comment]
Hi @robocoder , thanks for sharing this information! Our team will look into it and get back to you when we know where the issue comes from.

**Comment by eugenels** [comment]
As for now, the `~` operator expects string literal as a second argument (regexp part), and cannot be parameterised with a prepared statement.

But it’s subject to change.
Function signature can be “relaxed” to accept both const (literal) values and “runtime const” values.

**Reply by robocoder** [reply]
Does "subject to change" mean there's a plan to change it? Is there a list of which operators can/can't be parameterized?
